ddl/ddl_api.go

```diff
@@ -1975,40 +1971,19 @@ func checkPartitionByRange(ctx sessionctx.Context, tbInfo *model.TableInfo, s *a
 	}
 
 	if len(pi.Columns) == 0 {
-		if err := checkCreatePartitionValue(ctx, tbInfo); err != nil {
-			return err
-		}
-
-		// s maybe nil when add partition.
```

Comment:
I don't get why we can remove the code here, can you explain it a little bit?

Reply:
Since that code was moved into another place to reduce the duplicate code.